eclipse-theia/theia
# ci: fix `set-milestone-on-pr` workflow

#### What it does


Fixes: https://github.com/eclipse-theia/theia/issues/10630.

The commit fixes an issue with the `set-milestone-on-pr` workflow which did not work.
The change includes using `semver` to compute the next version correctly.

#### How to test


1. checkout the pull-request
2. ensure that the pull-request is not built - `git clean -ffdx`
3. perform `cd packages/workspace` 
4. execute:
    ```
    $ (
       export THEIA_CORE_VERSION=$(node -p "require(\"./packages/core/package.json\").version")
       echo "MILESTONE_NUMBER=$(npx -q semver@7 --increment minor $THEIA_CORE_VERSION)"
    )
    ```
5. confirm that the `MILESTONE_NUMBER` works

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>
